python/plugins/processing/algs/gdal/gdaltindex.py

```diff
@@ -162,6 +155,11 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
             arguments.append('-f {}'.format(outFormat))
 
         arguments.append(output)
-        arguments.append(' '.join(layers))
+
+        # Always write input files to a text file in case there are many of them and the
+        # length of the command will be longer then allowed in command prompt
+        list_file = GdalUtils.writeLayerParameterToTextFile(filename='tile_index_files.txt', alg=self, parameters=parameters, parameter_name=self.LAYERS, context=context, quote=True, executing=executing)
```

Comment:
make sense to create a filename based on session or uuid to avoid overlap in case of parallell executions?

Reply:
Fixed